lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/EvaluatorManager.cs

```diff
@@ -217,11 +217,25 @@ internal void RecordFailedEvaluator(string evaluatorId)
         }
 
         /// <summary>
+        /// Remove failed evaluator from the colletion
+        /// </summary>
+        /// <param name="evaluatorId"></param>
+        internal void RemoveFailedEvaluator(string evaluatorId)
+        {
+            if (!_failedEvaluatorIds.Contains(evaluatorId))
+            {
+                string msg = string.Format("The failed evaluator {0} is not recorded.", evaluatorId);
+                Exceptions.Throw(new IMRUSystemException(msg), Logger);
+            }
+            _failedEvaluatorIds.Remove(evaluatorId);
+        }
+
+        /// <summary>
         /// Checks if the number of failed Evaluators has reached allowed maximum number of evaluator failures 
         /// </summary>
         internal bool ReachedMaximumNumberOfEvaluatorFailures()
```

Comment:
I feel `HasExceededMaxNumberOfEvaluatorFailures` would be a better name. "Reached" sounds to me as "up to a limit", while the method in fact checks if the number of failed evaluators is greater than the limit.